syncer/syncer.go

```diff
@@ -163,31 +177,61 @@ func (s *syncer) BulkSync(ctx context.Context, newBlockCallback func(*types.Bloc
 }
 
 func (s *syncer) WatchSync(ctx context.Context, callback func(*types.Block) bool) error {
+	localLatest := s.blockchain.Header().Number
+	skipList := make(map[peer.ID]bool)
+
 	// Loop until context is canceled
 	for {
+		select {
+		case <-s.newStatus:
+		case <-time.After(s.blockTimeout):
+			return errTimeout
+		}
+
+		// fetch local latest block
+		if header := s.blockchain.Header(); header != nil {
+			localLatest = header.Number
+		}
+
 		// pick one best peer
-		// peer := s.peerHeap.BestPeer()
+		bestPeer := s.peerMap.BestPeer(skipList)
+		if bestPeer == nil {
+			break
+		}
+
+		// if the bestPeer does not have a new block continue
+		if bestPeer.Number <= localLatest {
+			continue
+		}
 
 		// fetch block from the peer
+		lastNumber, isValidator, err := s.bulkSyncWithPeer(bestPeer.ID, callback)
```

Comment:
`isValidator` is not proper name because there is no relationship between syncer and validator.

Reply:
Fixed, what do you think about this naming 3d198079c12bfe45e2b7f54e4aae7b9fddbea315